specification/compute/resource-manager/Microsoft.Compute/preview/2020-09-30/gallery.json

```diff
@@ -0,0 +1,2836 @@
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/GalleryApplicationList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List gallery Applications in a gallery.": {
+            "$ref": "./examples/ListGalleryApplicationsInAGallery.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/applications/{galleryApplicationName}/versions": {
+      "get": {
+        "tags": [
+          "GalleryApplicationVersions"
+        ],
+        "operationId": "GalleryApplicationVersions_ListByGalleryApplication",
+        "description": "List gallery Application Versions in a gallery Application Definition.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "galleryName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Shared Application Gallery in which the Application Definition resides."
+          },
+          {
+            "name": "galleryApplicationName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Shared Application Gallery Application Definition from which the Application Versions are to be listed."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/GalleryApplicationVersionList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List gallery Application Versions in a gallery Application Definition.": {
+            "$ref": "./examples/ListGalleryApplicationVersionsInAGalleryApplication.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/galleries/{galleryName}/share": {
```

Comment:
The sharingProfile is present on the gallery, why does it need a POST action to make changes to it?

Reply:
From our design, sharing profile is new and critical resource. Only this uri can be used for POST gallery and change gallery type from 'Private' to 'Group' and update groups. For the gallery operation, 'permissions' is a required property but 'groups' is an output-only field.

Reply:
This prevents users from sharing their galleries via a template deployment.  Please ensure you've thought through the impact that may have to customers

Reply:
Yes. The design is that user can not share to others when doing gallery resource deployment. We see sharing profile(sharing group) as another critical resource and want to use this api for user to do the sharing part.